### PR TITLE
Drop support php 7, add php 8.2 support

### PR DESCRIPTION
**Pull request by fezfez**

### CODE REVIEW

**Comment by Ocramius** [comment]
We'll need a PHPUnit and PHPCS upgrade here

**Comment by fezfez** [comment]
> We'll need a PHPUnit and PHPCS upgrade here

OK :)

**Comment by fezfez** [comment]
@Ocramius : ci OK

File: src/Decoder.php [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare(strict_types=1);
```
Comment (Ocramius) [inline src/Decoder.php]: Hmm, introducing this is problematic, if not a BC break.

I suggest disabling the rule, like we did, for example, in https://github.com/laminas/laminas-validator/blob/23e27edc1520c0bf0c9c7e7f192dd1314677969e/phpcs.xml#L28-L30

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Decoder.php (hunk 1 above)
Comment (fezfez) [reply src/Decoder.php]: Done

**Review by Ocramius — APPROVED** [review]
Thanks @fezfez!

**Pull request merged**